### PR TITLE
PLAT-647: A.New with twice pulse change in ExecutionStart

### PR DESCRIPTION
**Pull request by ElSacramento**
add test for twice pulse change during constructor execution
```
$ go test ./virtual/integration/ -test.run ^TestVirtual_CallConstructor_WithTwicePulseChange$ -count=1000 -race
ok  	github.com/insolar/assured-ledger/ledger-core/virtual/integration	24.344s
$ go test ./virtual/integration/ -test.run ^TestVirtual_CallConstructor_WithTwicePulseChange$ -count=1000
ok  	github.com/insolar/assured-ledger/ledger-core/virtual/integration	4.827s
```

### CODE REVIEW

File: ledger-core/virtual/integration/constructor_test.go [hunk 1]
```diff
@@ -804,6 +805,171 @@ func TestVirtual_Constructor_PulseChangedWhileOutgoing(t *testing.T) {
...
+				t.Fatal("unexpected")
+			}
+
+			server.SendPayload(ctx, &msg)
+			return false
+		})
+		typedChecker.VDelegatedRequestFinished.Set(func(finished *payload.VDelegatedRequestFinished) bool {
+			assert.Equal(t, objectRef, finished.Callee)
+			assert.Equal(t, secondExpectedToken, finished.DelegationSpec)
+			return false
```
Comment (ruz) [inline ledger-core/virtual/integration/constructor_test.go:921]: let's check memory here

File: ledger-core/virtual/integration/constructor_test.go [hunk 2]
```diff
@@ -804,6 +805,171 @@ func TestVirtual_Constructor_PulseChangedWhileOutgoing(t *testing.T) {
...
+
+	synchronizeExecution := synchronization.NewPoint(1)
+
+	// add ExecutionMocks to runnerMock
+	{
+		objectAResult := requestresult.New([]byte("finish A.New"), outgoing)
+		objectAResult.SetActivate(reference.Global{}, classA, []byte("state A"))
+		runnerMock.AddExecutionMock("New").AddStart(
+			func(_ execution.Context) {
+				server.IncrementPulseAndWaitIdle(ctx)
```
Comment (ruz) [inline ledger-core/virtual/integration/constructor_test.go:860]: can you please move pulse changes from here to the block at the end? Leave here `synchronizeExecution.Synchronize()`. At the end in a block do the following: `syncPoint.Wait()`, `IncrementPulseAndWaitIdle()`,  `IncrementPulseAndWaitIdle()`, `syncPoint.Release()`

Comment (ElSacramento) [reply ledger-core/virtual/integration/constructor_test.go]: done

File: ledger-core/virtual/integration/constructor_test.go (hunk 1 above)
Comment (ElSacramento) [reply ledger-core/virtual/integration/constructor_test.go:919]: done

File: ledger-core/virtual/integration/constructor_test.go [hunk 3]
```diff
@@ -819,6 +820,171 @@ func TestVirtual_Constructor_PulseChangedWhileOutgoing(t *testing.T) {
...
+				predicate.NewSMTypeFilter(&execute.SMExecute{}, predicate.BeforeStep((&execute.SMExecute{}).StepWaitExecutionResult)),
+			),
+		)
+		// wait for first pulse change
+		server.IncrementPulseAndWaitIdle(ctx)
+		testutils.WaitSignalsTimed(t, 20*time.Second, tokenRequestDone)
+
+		// wait for second pulse change
+		server.IncrementPulseAndWaitIdle(ctx)
+		testutils.WaitSignalsTimed(t, 20*time.Second, tokenRequestDone)
```
Comment (bigbes) [inline ledger-core/virtual/integration/constructor_test.go:972]: Nope, once Wait is done it won't be checked second time. Please, check it using for loop and i < 2

Comment (ElSacramento) [reply ledger-core/virtual/integration/constructor_test.go]: done

File: ledger-core/virtual/integration/constructor_test.go [hunk 4]
```diff
@@ -819,6 +820,169 @@ func TestVirtual_Constructor_PulseChangedWhileOutgoing(t *testing.T) {
...
+// -> change pulse -> secondPulse
+// -> VStateReport [A]
+// -> VDelegatedCallRequest [A]
+// -> change pulse -> thirdPulse
+// -> NO VStateReport
+// -> VDelegatedCallRequest [A] + first token
+// -> VCallResult [A.New] + second token
+// -> VDelegatedRequestFinished [A] + second token
+func TestVirtual_CallConstructor_WithTwicePulseChange(t *testing.T) {
+	t.Log("C5208")
```
Comment (sexyhamster) [inline ledger-core/virtual/integration/constructor_test.go:834]: add leackchecker please

Comment (ElSacramento) [reply ledger-core/virtual/integration/constructor_test.go:834]: done

**Pull request merged**